common/lib/xmodule/xmodule/tabs.py

```diff
@@ -789,43 +208,47 @@ def initialize_default(course):
         """
 
         course.tabs.extend([
-            CoursewareTab(),
-            CourseInfoTab(),
+            CourseTab.from_json({'type': 'courseware', 'name': _('Courseware')}),
+            CourseTab.from_json({'type': 'course_info', 'name': _('Course Info')})
         ])
 
         # Presence of syllabus tab is indicated by a course attribute
         if hasattr(course, 'syllabus_present') and course.syllabus_present:
-            course.tabs.append(SyllabusTab())
+            course.tabs.append(CourseTab.from_json({'type': 'syllabus', 'name': _('Syllabus')}))
 
         # If the course has a discussion link specified, use that even if we feature
         # flag discussions off. Disabling that is mostly a server safety feature
         # at this point, and we don't need to worry about external sites.
         if course.discussion_link:
-            discussion_tab = ExternalDiscussionTab(link_value=course.discussion_link)
+            discussion_tab = CourseTab.from_json(
+                {'type': 'external_discussion', 'name': _('External Discussion'), 'link': course.discussion_link}
+            )
         else:
-            discussion_tab = DiscussionTab()
+            discussion_tab = CourseTab.from_json({'type': 'discussion', 'name': _('Discussion')})
 
         course.tabs.extend([
-            TextbookTabs(),
+            CourseTab.from_json({'type': 'textbooks', 'name': _('Textbooks')}),
             discussion_tab,
-            WikiTab(),
-            ProgressTab(),
+            CourseTab.from_json({'type': 'wiki', 'name': _('Wiki')}),
+            CourseTab.from_json({'type': 'progress', 'name': _('Progress')}),
         ])
 
     @staticmethod
     def get_discussion(course):
         """
-        Returns the discussion tab for the given course.  It can be either of type DiscussionTab
-        or ExternalDiscussionTab.  The returned tab object is self-aware of the 'link' that it corresponds to.
+        Returns the discussion tab for the given course.  It can be either of type 'discussion'
+        or 'external_discussion'.  The returned tab object is self-aware of the 'link' that it corresponds to.
         """
 
         # the discussion_link setting overrides everything else, even if there is a discussion tab in the course tabs
         if course.discussion_link:
```

Comment:
I know this code hasn't changed, but why does it return a dynamically created tab that isn't in the course? Wouldn't the external discussion tab already have been created? It seems risky to return a tab that isn't actually being shown.